cloudformation/synthetic-event-generation.yaml

```diff
@@ -0,0 +1,104 @@
+      KmsKeyArn: !GetAtt KmsKey.Arn
+      Timeout: 180
+      Policies:
+        - AWSLambdaVPCAccessExecutionRole
+        - Statement:
+            - Effect: Allow
+              Action:
+                - kms:Decrypt
+                - kms:GenerateDataKey*
+                - kms:Encrypt
+              Resource: !GetAtt KmsKey.Arn
+            - Effect: Allow
+              Action:
+                - kms:Decrypt
+                - kms:GenerateDataKey*
+                - kms:Encrypt
+              Resource: '*'
+              Condition:
+                ForAnyValue:StringEquals:
+                  kms:ResourceAliases: !Sub alias/key-${ConfigStackParameter.Value}-general
+            - Effect: Allow
+              Action:
+                - s3:AbortMultipartUpload
+                - s3:GetBucketLocation
+                - s3:GetObject
+                - s3:ListBucket
+                - s3:ListBucketMultipartUploads
+                - s3:ListMultipartUploadParts
+                - s3:PutObject
+              Resource:
+                - !Sub arn:aws:s3:::${ConfigStackParameter.Value}
+                - !Sub arn:aws:s3:::${ConfigStackParameter.Value}/*
+            - Effect: Allow
+              Action:
+                - sqs:SendMessage
+              Resource: !GetAtt StorageQueue.Arn
+      ReservedConcurrentExecutions: 1
+
+  SyntheticEventGenerationScheduledRule:
+    Type: AWS::Events::Rule
+    Properties:
+      Name: !Sub ${AWS::StackName}-synthetic-event-generation-rule
+      Description: Schedules SyntheticEventGenerationFunction to run once per day
+      ScheduleExpression: cron(0 10 1 * ? *) # Schedule for 10 a.m. on the first of every month
+      State: ENABLED
+      Targets:
+        - Arn: !GetAtt SyntheticEventGenerationFunction.Arn
+          Id: SyntheticEventGenerationFunction
+
+  LambdaPermissionForEvents:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Ref SyntheticEventGenerationFunction
+      Action: lambda:InvokeFunction
+      Principal: events.amazonaws.com
+      SourceArn: !GetAtt SyntheticEventGenerationScheduledRule.Arn
+
+  LambdaPermissionForS3SyntheticEventGenerationFunction:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Ref SyntheticEventGenerationFunction
+      Action: lambda:InvokeFunction
+      Principal: s3.amazonaws.com
+      SourceArn: !Sub arn:aws:s3:::${AWS::StackName}-storage
+
+  SyntheticEventGenerationFunctionAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: !Sub ${AWS::StackName}-synthetic-event-generation-function-alarm
+      AlarmActions:
+        - !GetAtt AlarmTopicParameter.Value
+      ComparisonOperator: GreaterThanThreshold
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref SyntheticEventGenerationFunction
+      EvaluationPeriods: 1
+      MetricName: Errors
+      Namespace: AWS/Lambda
+      Period: 60
+      Statistic: Sum
+      Threshold: 0
```

Comment:
Hi @markschnitziusgovuk  noticed there is no DLQ for this lambda so wanted to understand how we currently handle event retries for this lambda.